coreax/coreset.py

```diff
@@ -0,0 +1,114 @@
+"""Module for defining coreset data structures."""
+
+from __future__ import annotations
+
+import equinox as eqx
+import jax.numpy as jnp
+from jaxtyping import Array, Shaped
+
+from coreax.data import Data
+
+
+def _convert_to_weighted(data: Data | Array):
+    if isinstance(data, Data):
+        return data
+    return Data(data)
+
+
+class Coreset(eqx.Module):
+    r"""
+    Data structure for representing a coreset.
+
+    TLDR: a coreset is a reduced set of :math:`\hat{n}` (potentially weighted) data
+    points that, in some sense, best represent the "important" properties of a larger
+    set of :math:`n > \hat{n}` (potentially weighted) data points.
```

Comment:
Could stick in a >>> here to signify that the coreset is supposed to be much smaller than n

Reply:
It depends if you would give a set where `n` is with `>` but not `>>` `\hat{n}` a different name? If not, then I think the current notation is more accuracte.

Reply:
I guess philosophically if you've got n=1,000,000 and hat(n)=999,999 then its not much of a coreset, but i take your point and happy to leave as is!